.emacs.d/init.el

```diff
@@ -109,6 +116,10 @@ Despite this, your config appears to have loaded successfully.")
   (if loaddefs-buffer
       (kill-buffer loaddefs-buffer)))
 
+;; If in dev mode, load the test runner.
+(when kotct/dev-mode
+  (require 'dot-tests "~/.emacs.d/test/lisp/dot-tests"))
```

Comment:
>It adds calls to the `init` stream under normal non-dev circumstances

Hardly. I wouldn't consider a variable definition and an if-statement to be adding anything.

>Adding this to the source tree for the entire configuration means that we're tainting our actual configuration with testing stuff.

I mean... the tests do not run or actually do anything unless you explicitly run them, so the only "tainting" is the definition of a few functions. Also this only happens in dev mode.

>I think tests should be run in a separate Emacs instance from the existing one, if the developer is running one. The ability to run `kotct/run-tests` within the normal running circumstances is somewhat sketchy to me.

Officially, a separate Emacs instance is the correct way to run the tests. If `emacs --script ~/.emacs.d/test/lisp/run-tests.el` fails, the tests do not pass. That is how they're run in Travis. That said, having the ability to run the tests in your Emacs session is a convenience that makes it much easier to run/debug tests, and there's very little overhead.